mobile/lib/services/backup.service.dart

```diff
@@ -81,54 +92,48 @@ class BackupService {
       imageOption: const FilterOption(needTitle: true),
       videoOption: const FilterOption(needTitle: true),
     );
-    final now = DateTime.now();
+
     final List<AssetPathEntity?> selectedAlbums =
-        await _loadAlbumsWithTimeFilter(selectedBackupAlbums, filter, now);
+        await _loadAlbumsWithTimeFilter(
+      selectedBackupAlbums,
+      filter,
+    );
+
     if (selectedAlbums.every((e) => e == null)) {
-      return [];
-    }
-    final int allIdx = selectedAlbums.indexWhere((e) => e != null && e.isAll);
-    if (allIdx != -1) {
-      final List<AssetPathEntity?> excludedAlbums =
-          await _loadAlbumsWithTimeFilter(excludedBackupAlbums, filter, now);
-      final List<AssetEntity> toAdd = await _fetchAssetsAndUpdateLastBackup(
-        selectedAlbums.slice(allIdx, allIdx + 1),
-        selectedBackupAlbums.slice(allIdx, allIdx + 1),
-        now,
-      );
-      final List<AssetEntity> toRemove = await _fetchAssetsAndUpdateLastBackup(
-        excludedAlbums,
-        excludedBackupAlbums,
-        now,
-      );
-      return toAdd.toSet().difference(toRemove.toSet()).toList();
-    } else {
-      return await _fetchAssetsAndUpdateLastBackup(
-        selectedAlbums,
-        selectedBackupAlbums,
-        now,
-      );
+      return {};
     }
+
+    final List<AssetPathEntity?> excludedAlbums =
+        await _loadAlbumsWithTimeFilter(
+      excludedBackupAlbums,
+      filter,
+    );
+
+    final Set<BackupCandidate> toAdd = await _fetchAssetsAndUpdateLastBackup(
+      selectedAlbums,
+      selectedBackupAlbums,
+    );
+
+    final Set<BackupCandidate> toRemove = await _fetchAssetsAndUpdateLastBackup(
+      excludedAlbums,
+      excludedBackupAlbums,
+    );
+
+    return toAdd.difference(toRemove);
   }
 
   Future<List<AssetPathEntity?>> _loadAlbumsWithTimeFilter(
     List<BackupAlbum> albums,
     FilterOptionGroup filter,
-    DateTime now,
   ) async {
     List<AssetPathEntity?> result = [];
-    for (BackupAlbum a in albums) {
+
+    for (BackupAlbum backupAlbum in albums) {
       try {
         final AssetPathEntity album =
             await AssetPathEntity.obtainPathFromProperties(
-          id: a.id,
-          optionGroup: filter.copyWith(
-            updateTimeCond: DateTimeCond(
```

Comment:
Why is this removed?
This will greatly increase the time of the background upload each time it's executed for large libraries leading to not uploading stuff at all on iOS and wasting battery on Android as it's triggered very often (after any assets changed on device)

Reply:
I found that with this filtering condition, and with the scenario of a fresh installed, and choosing let's say "Camera" and "Recent" album for backup. When running in background mode, only the "Recent" album gets processed for the included albums. 

Any thoughts?

Reply:
I'm not sure, but something in this PR might have changed the logic. It works fine for multiple albums and backs up all assets on main/release.

I believe this requires some debugging (has the album modified date changed: yes/no, are less assets obtained from the album than expected: yes/no, ...).
It's weird that only the background backup is affected - as the code is identical.

Reply:
Ah, it was an error in the conditional parameter that was unused. It is now put back and work as expected